server/api/src/rsGenQuery.cpp

```diff
@@ -214,7 +212,33 @@ namespace {
         }
 
         return fmt::format("IN ({})", fmt::join(leaf_ids, ","));
-    } // get_resc_id_cond_for_hier_cond
+    } // translate_single_data_resc_hier_condition_to_resc_id
+
+    auto translate_data_resc_hier_where_clause_to_resc_id(const std::string& _condition) -> std::string
+    {
+        std::string condition_str;
+
+        // For each || operator found, take the condition immediately preceding and translate it from the DATA_RESC_HIER
+        // condition to the equivalent RESC_ID condition.
+        auto previous_or_position = std::string_view::size_type{};
+        for (auto or_position = _condition.find("||", previous_or_position); std::string_view::npos != or_position;
+             or_position = _condition.find("||", previous_or_position))
+        {
+            condition_str += translate_single_data_resc_hier_condition_to_resc_id(
+                _condition.substr(previous_or_position, or_position));
+            condition_str += " || ";
+
+            // Advance the previous position tracker to the current || position, plus 2 so that the next condition to
+            // translate does not include the || itself.
+            previous_or_position = or_position + 2;
+        }
+
+        // There should be one condition found after the last || operator has been found which will need to be
+        // translated because the loop above is translating conditions BEFORE the || operator. || is a binary operation,
+        // so there will necessarily be one left over at the end.
+        condition_str += translate_single_data_resc_hier_condition_to_resc_id(_condition.substr(previous_or_position));
+        return condition_str;
```

Comment:
This can be one line because `+=` returns the object on the left for `std::string`.

Feel free to ignore as this is perfectly acceptable.

Reply:
Will do